src/main/java/bit/javaoop/Company.java

```diff
@@ -0,0 +1,108 @@
+package bit.javaoop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Company {
+
+    private String name;
+    private Ceo ceo;
+    private List<AbstractManager> managersList = new ArrayList<>();
+    private List<AbstractEmployee> employeesList = new ArrayList<>();
+
+    public Company(String name) {
+        this.name = name;
+    }
+
+    public Company(String name, Ceo ceo) {
+        this.name = name;
+        this.ceo = ceo;
+    }
+
+
+    public void setCeo(Ceo ceo) {
+        if (this.ceo != null) {
+            this.ceo.setManagersList(null);
```

Comment:
Ciężko się od razu domyślić co to ma robić, proponowałbym stworzenie nowej metody w klasie Ceo, która by czyściła listę Manager'ów.